aci/resource_aci_dhcprelayp.go

```diff
@@ -0,0 +1,339 @@
+	desc := d.Get("description").(string)
+
+	name := d.Get("name").(string)
+
+	TenantDn := d.Get("tenant_dn").(string)
+
+	dhcpRelayPAttr := models.DHCPRelayPolicyAttributes{}
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		dhcpRelayPAttr.Annotation = Annotation.(string)
+	} else {
+		dhcpRelayPAttr.Annotation = "{}"
+	}
+	if Mode, ok := d.GetOk("mode"); ok {
+		dhcpRelayPAttr.Mode = Mode.(string)
+	}
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		dhcpRelayPAttr.NameAlias = NameAlias.(string)
+	}
+	if Owner, ok := d.GetOk("owner"); ok {
+		dhcpRelayPAttr.Owner = Owner.(string)
+	}
+	dhcpRelayP := models.NewDHCPRelayPolicy(fmt.Sprintf("relayp-%s", name), TenantDn, desc, dhcpRelayPAttr)
+
+	err := aciClient.Save(dhcpRelayP)
+	if err != nil {
+		return err
+	}
+	d.Partial(true)
+
+	d.SetPartial("name")
+
+	d.Partial(false)
+
+	checkDns := make([]string, 0, 1)
+
+	if relationTodhcpRsProv, ok := d.GetOk("relation_dhcp_rs_prov"); ok {
+		relationParam := relationTodhcpRsProv.(string)
+		checkDns = append(checkDns, relationParam)
+	}
+
+	d.Partial(true)
+	err = checkTDn(aciClient, checkDns)
+	if err != nil {
+		return err
+	}
+	d.Partial(false)
+
+	if relationTodhcpRsProv, ok := d.GetOk("relation_dhcp_rs_prov"); ok {
+		relationParamList := toStringList(relationTodhcpRsProv.(*schema.Set).List())
+		for _, relationParam := range relationParamList {
+			err = aciClient.CreateRelationdhcpRsProvFromDHCPRelayPolicy(dhcpRelayP.DistinguishedName, relationParam)
+
+			if err != nil {
+				return err
+			}
+			d.Partial(true)
+			d.SetPartial("relation_dhcp_rs_prov")
+			d.Partial(false)
+		}
+	}
+
+	d.SetId(dhcpRelayP.DistinguishedName)
+	log.Printf("[DEBUG] %s: Creation finished successfully", d.Id())
+
+	return resourceAciDHCPRelayPolicyRead(d, m)
+}
+
+func resourceAciDHCPRelayPolicyUpdate(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] DHCPRelayPolicy: Beginning Update")
+
+	aciClient := m.(*client.Client)
+	desc := d.Get("description").(string)
+
+	name := d.Get("name").(string)
+
+	TenantDn := d.Get("tenant_dn").(string)
+
+	dhcpRelayPAttr := models.DHCPRelayPolicyAttributes{}
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		dhcpRelayPAttr.Annotation = Annotation.(string)
+	} else {
+		dhcpRelayPAttr.Annotation = "{}"
+	}
+	if Mode, ok := d.GetOk("mode"); ok {
+		dhcpRelayPAttr.Mode = Mode.(string)
+	}
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		dhcpRelayPAttr.NameAlias = NameAlias.(string)
+	}
+	if Owner, ok := d.GetOk("owner"); ok {
+		dhcpRelayPAttr.Owner = Owner.(string)
+	}
+	dhcpRelayP := models.NewDHCPRelayPolicy(fmt.Sprintf("relayp-%s", name), TenantDn, desc, dhcpRelayPAttr)
+
+	dhcpRelayP.Status = "modified"
+
+	err := aciClient.Save(dhcpRelayP)
+
+	if err != nil {
+		return err
+	}
+	d.Partial(true)
+
+	d.SetPartial("name")
+
+	d.Partial(false)
+
+	checkDns := make([]string, 0, 1)
+
+	if d.HasChange("relation_dhcp_rs_prov") {
+		_, newRelParam := d.GetChange("relation_dhcp_rs_prov")
```

Comment:
This relation has list of dns. So update the logic here for accepting multiple strings.